docs/modules/ROOT/pages/reference/architecture.adoc

```diff
@@ -299,6 +303,52 @@ Commodore currently doesn't provide support for component authors to specify lib
 It's the responsibility of component authors to agree on an interface and to ensure that their implementations adhere to the interface.
 ====
 
+=== Component instance versions
```

Comment:
I'd move this section above `=== Component dependencies` so it comes directly after `=== Component instances`

Note that I've renamed it to better match the other section titles